<a href="https://colab.research.google.com/github/a3hok/Data-Visualization-spring-2022/blob/main/quiz5.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

Quiz 5

Ashok Dahal

##SQL SubQuery



The following command is for Authenticate dataset from google cloud platform

In [1]:
from google.colab import auth
auth.authenticate_user()
print('Authenticated')

Authenticated


Plotting the dataset of austin bikeshare in the google cloud platform using BigQuery.


In [3]:
%%bigquery --project data-visualization-339004
SELECT 
  COUNT(*) as total_no_of_entries
FROM `bigquery-public-data.austin_bikeshare.bikeshare_trips`

total_no_of_entries
0              1424786

1,424,786 number of entries are there in the Austine Bikeshare dataset of google cloud platform



#Finding out second most popular starting station.

In [5]:
%%bigquery --project data-visualization-339004
SELECT start_station_name, COUNT(*) as second_popular
FROM `bigquery-public-data.austin_bikeshare.bikeshare_trips`
GROUP BY start_station_name
HAVING second_popular!= (SELECT MAX(second_popular)as max_trip
FROM (
SELECT COUNT(*) as second_popular, start_station_name
FROM `bigquery-public-data.austin_bikeshare.bikeshare_trips`
GROUP BY start_station_name
))
ORDER BY second_popular DESC
LIMIT 1

start_station_name  second_popular
0  Riverside @ S. Lamar           40635

 The above is the second most popular starting station

#computing the round trip which started and completed at the same station.

In [11]:
%%bigquery --project data-visualization-339004

SELECT start_station_name, duration_minutes
FROM `bigquery-public-data.austin_bikeshare.bikeshare_trips`
WHERE start_station_name = end_station_name

start_station_name  duration_minutes
0       Toomey Rd @ South Lamar                82
1              Waller & 6th St.                76
2              Waller & 6th St.                75
3              Waller & 6th St.                74
4              Waller & 6th St.                73
...                         ...               ...
228334                 3rd/West                63
228335               3rd & West                63
228336                 3rd/West                63
228337                 3rd/West                63
228338              26th/Nueces                63

[228339 rows x 2 columns]

#computing trips lasted over an hour and were a round trip (started and stopped at the same station)

In [10]:
%%bigquery --project data-visualization-339004

SELECT *, COUNTIF(duration_minutes>60) as trip_over_one_hour
FROM (
  SELECT start_station_name, duration_minutes
FROM `bigquery-public-data.austin_bikeshare.bikeshare_trips`
WHERE start_station_name = end_station_name
)
GROUP BY duration_minutes, start_station_name
Order BY trip_over_one_hour DESC

start_station_name  duration_minutes  trip_over_one_hour
0      Riverside @ S. Lamar                62                 109
1      Riverside @ S. Lamar                61                  95
2      Riverside @ S. Lamar                63                  94
3      Riverside @ S. Lamar                69                  90
4               Zilker Park                61                  88
...                     ...               ...                 ...
24822           26th/Nueces                40                   0
24823           26th/Nueces                17                   0
24824           26th/Nueces                53                   0
24825           26th/Nueces                48                   0
24826           26th/Nueces                18                   0

[24827 rows x 3 columns]